ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/tasks/CompleteBlocksTask.java

```diff
@@ -79,7 +79,11 @@ private CompleteBlocksTask(
 
   private boolean hasEmptyBody(final BlockHeader header) {
     return header.getOmmersHash().equals(Hash.EMPTY_LIST_HASH)
-        && header.getTransactionsRoot().equals(Hash.EMPTY_TRIE_HASH);
+        && header.getTransactionsRoot().equals(Hash.EMPTY_TRIE_HASH)
+        && header
+            .getWithdrawalsRoot()
```

Comment:
does it though?  if withdrawals are not enabled wouldn't we get an empty Optional?  This looks correct to me.  

I presume we already have test coverage that asserts we will fail a block that tries to execute withdrawals before shanghai.

Reply:
We discussed on a call. Decided it wasn't needed due to later validation as you say, but we do need to check the protocolSchedule for if withdrawals is enabled to be able to create a withdrawals-aware empty block. That changes and unit tests incoming

Reply:
Yeah it didn't look right when I first looked at it. But we had a closer look at your right we don't need to do that check here as the validation will take care of that. It would be moving some of the validation into the class which isn't right.